resolve "prefer-arrow-callback" violations in javascripts/fetchIssueCount.js

We have some very old ESLint rules that I'd like to clean up, and our use of `var` is one of them. ES6 introduces `let` and `const` as a way to make clearer when a variable can be used, and I'd like to reduce our use of `var` where we don't need it:

If you know a bit about JS you can help out by following these steps:

 - create a new branch from `gh-pages`
 - in `javascripts/fetchIssueCount.js` add this comment at the top of the file, before any source code:

```js
/* eslint prefer-arrow-callback: [ "error" ] */
```

 - in a terminal, run `npm run lint`, which should emit a bunch of violations:

```
$ npm run lint

> up-for-grabs.net@1.0.0 lint /Users/shiftkey/src/up-for-grabs.net
> eslint javascripts/*.js tests/**/*.js


/Users/shiftkey/src/up-for-grabs.net/javascripts/fetchIssueCount.js
   11:46  error  Unexpected function expression  prefer-arrow-callback
  167:24  error  Unexpected function expression  prefer-arrow-callback
   ...
```

 - run `npm run lint-fix` to fix whatever can be automatically tidied up
 - review and address the remaining violations manually
 - run `npm run prettier-fix` to ensure the formatting of the changes is fine
 - [test the site locally](https://github.com/up-for-grabs/up-for-grabs.net#testing-the-site-locally)
 - commit the changes and open a PR
 - test out the Netlify deploy site (click the `Details` link) and confirm the client-side functionality is still working

<img width="709" src="https://user-images.githubusercontent.com/359239/66651591-c70eaf80-ec09-11e9-85bc-a8c58d81e2be.png">

--- a/javascripts/fetchIssueCount.js
+++ b/javascripts/fetchIssueCount.js
@@ -1,5 +1,6 @@
 /* eslint global-require: "off" */
 /* eslint block-scoped-var: "off" */
+/* eslint prefer-arrow-callback: [ "error" ] */
 
 // @ts-nocheck
 
@@ -8,7 +9,7 @@ if (typeof define !== 'function') {
   var define = require('amdefine')(module);
 }
 
-define(['whatwg-fetch', 'promise-polyfill'], function() {
+define(['whatwg-fetch', 'promise-polyfill'], () => {
   const { localStorage, fetch } = window;
 
   const RateLimitResetAtKey = 'Rate-Limit-Reset-At';
@@ -164,9 +165,9 @@ define(['whatwg-fetch', 'promise-polyfill'], function() {
       };
     }
 
-    return new Promise(function(resolve, reject) {
-      return fetch(apiURL, settings).then(
-        function(response) {
+    return new Promise((resolve, reject) =>
+      fetch(apiURL, settings).then(
+        response => {
           if (!response.ok) {
             if (response.status === 304) {
               // no content is returned in the 304 Not Modified response body
@@ -184,10 +185,10 @@ define(['whatwg-fetch', 'promise-polyfill'], function() {
             }
 
             response.json().then(
-              function(json) {
+              json => {
                 reject(inspectGenericError(json, response));
               },
-              function(error) {
+              error => {
                 reject(error);
               }
             );
@@ -219,7 +220,7 @@ define(['whatwg-fetch', 'promise-polyfill'], function() {
           }
 
           response.json().then(
-            function(json) {
+            json => {
               if (json && typeof json.length === 'number') {
                 const count = json.length;
                 setValue(ownerAndName, {
@@ -231,16 +232,16 @@ define(['whatwg-fetch', 'promise-polyfill'], function() {
                 resolve(count);
               }
             },
-            function(error) {
+            error => {
               reject(error);
             }
           );
         },
-        function(error) {
+        error => {
           reject(error);
         }
-      );
-    });
+      )
+    );
   }
 
   return fetchIssueCount;

--- a/javascripts/fetchIssueCount.js
+++ b/javascripts/fetchIssueCount.js
@@ -1,6 +1,9 @@
 /* eslint global-require: "off" */
 /* eslint block-scoped-var: "off" */
 /* eslint prefer-arrow-callback: [ "error" ] */
+/* eslint arrow-parens: [ "error", "as-needed" ] */
+/* eslint function-paren-newline: [ "off" ] */
+/* eslint implicit-arrow-linebreak: [ "off" ] */
 
 // @ts-nocheck
 

